src/main/java/io/vertx/circuitbreaker/impl/CircuitBreakerMetrics.java

```diff
@@ -0,0 +1,249 @@
+package io.vertx.circuitbreaker.impl;
+
+import io.vertx.circuitbreaker.CircuitBreakerOptions;
+import io.vertx.core.Vertx;
+import io.vertx.core.impl.VertxInternal;
+import io.vertx.core.json.JsonObject;
+import org.HdrHistogram.Histogram;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Circuit breaker metrics.
+ *
+ * @author <a href="http://escoffier.me">Clement Escoffier</a>
+ */
+public class CircuitBreakerMetrics {
```

Comment:
Yes, even if callbacks should be called by the context thread, it may happen it's not.... We had a question about using the circuit breaker outside of the Vert.x context.